src/scatter-plot.js

```diff
@@ -54,6 +54,20 @@ dc.scatterPlot = function (parent, chartGroup) {
     var _nonemptyOpacity = 1;
     var _emptyColor = null;
     var _filtered = [];
+    var _canvas = null;
+    var _context = null;
+    var _useCanvas = false;
+
+    // Calculates element radius for canvas plot to be comparable to D3 area based symbol sizes
+    function canvasElementSize (d, isFiltered) {
+        if (!_existenceAccessor(d)) {
+            return _emptySize / Math.sqrt(Math.PI);
```

Comment:
I was surprised by `Math.sqrt(Math.PI)` here. How did you ever figure this out? 

I checked in the debugger and yes that's the factor by which the `d3.svg.symbol`-generated arc differs from the radius. 

I always thought they were coming out too small!

I bet the assumption in `elementSize` that we should square the radius in order to get `d3.svg.symbol.size` is wrong. So it's easy to see being off by a factor of π but the sqrt is really confusing.

Reply:
Haha, yeah, I had the same issue initially. I believe it is because the D3 docs mention that the supplied symbol size should specify the area of the symbol, not the radius. 

For circles, since the area is `Math.PI * (r^2)` when we define the radius of the symbol in dc.js, the method [`elementSize()`](https://github.com/HamsterHuey/dc.js/blob/6b9a40b8bf6ea2998f91ac4beb7ad04c1b53dfd0/src/scatter-plot.js#L72) should actually return `Math.pow(_emptySize, 2) * Math.PI`. But since it doesn't do this, the actual radius of the D3 symbol that gets plotted in the dc.js chart is off by a factor of `sqrt(Math.PI)`. Hence the correction by that factor.

I also believe that for all the other symbol types, D3 still relies on a circle based area calculation with the symbol extents guaranteed to lie within the circle defined by the provided area parameter. This block seems to support this guess - https://bl.ocks.org/mbostock/6d9d75ee13abbcfea6e0

So if other symbol types get implemented for canvas plots, the method for drawing the canvas symbol would need to make sure that each symbol type respects the perimeter defined by the circle.

Reply:
That example shows that the radius varies a lot for the same area depending on the shape.

So dc.js is definitely doing this wrong. The API names are correct but it should be using the size directly for `d3.svg.symbol` and not squaring it as if it's a radius (and getting the calculation wrong).

Would you be willing to fix the original problem and document the API change, so that the new code is not compounding the problem by introducing an obscure calculation in order to be compatible with an incorrect calculation? :smile:

API changes are fine in the 2.1 branch.

Reply:
Doh! Didn't notice that the example had identical symbol sizes. I just assumed that the way I had it in my head was how it would be implemented since that makes the most sense as the different symbols would then look to have a similar perceptual size for the same area parameter.

I don't mind implementing a fix that switches to size being defined as the area parameter as per the d3 docs for d3.symbol. Did you envision that being part of this PR or as a separate PR?